chainer/functions/loss/hinge_max_margin.py

```diff
@@ -0,0 +1,181 @@
+import numpy
+
+from chainer.backends import cuda
+from chainer import function as F
+from chainer.utils import type_check
+
+
+class HingeMaxMargin(F.Function):
```

Comment:
Could you inherit `FunctionNode` instead of `Function`?

Reply:
Yes, no problem